src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -800,15 +838,193 @@ def __try_add_error(error_list, detail):
+    def __create_truncated_status_file(self, status_file_size_in_bytes, complete_status_file_payload):
+        """ Truncate substatus message patch list when complete status file size is more than 126kb """
+        """
+            __create_truncated_status_file(self, status_file_size_in_bytes, complete_status_file_payload):
+
+                truncated_status_file = json.loads(complete_status_file_payload)
+                low_pri_index = None
+                _index = self.__get_substatus_index()
+                status_file_without_package_list_size  = __calc_package_payload_size_on_disk(size_of_constant_status_data(complete_status_file_payload))
+                size_of_max_packages_allowed_in_status = 126kb - status_file_without_package_list_size 
+                
+                if assessment_index is not none:
+                    _substatus_msg_copy = __get_substatus_message(truncated_status_file, _index)
+                    _packages_copy = _substatus_msg_copy['patches']
+
+                if installation_index is not none:
+                    _substatus_msg_copy = __get_substatus_message(truncated_status_file, _index)
+                    _packages_copy = _substatus_msg_copy['patches']
+                    low_pri_index = __get_installation_low_pri_index()
+
+                    while status_file_size_in_bytes > 126kb:
+                        __apply_truncation_process()
+                            __split_assessment_list()
+                                __apply_truncation()
+                                
+                        __recompose_truncated_status_file()                                    
+                            __get_current_complete_status_errors()
+                            __recompose_truncated_substatus_msg()
+                                __recompose_substatus_msg_errors()
+                                __create_assessment_tombstone_list()
+                                    __create_assessment_tombstone()
+                                __recreate_assessment_summary_json()
+                       
+                        __recompose_truncated_status_file()                                    
+                            __get_current_complete_status_errors()
+                            __recompose_truncated_substatus_msg()
+                                __recompose_substatus_msg_errors()
+                                __create_installation_tombstone
+                                __recreate_installation_summary_json()
+                                
+                        status_file_size_in_bytes, status_file_agent_size_diff = __get_new_size_in_bytes_after_truncation(truncated_status_file)
+                        size_of_max_packages_allowed_in_status -= status_file_agent_size_diff  
+        """
+        self.composite_logger.log_debug("Begin package list truncation")
+        truncated_status_file = json.loads(complete_status_file_payload)  # reload payload into python object
```

Comment:
Rename 'complete_status_file_payload' to 'complete_status_file_payload_json'

Reply:
done

Reply:
@rane-rajasi to resolve if satisfied.